protocols/imap/src/test/java/org/apache/james/imap/processor/GetAnnotationProcessorTest.java

```diff
@@ -205,29 +208,100 @@ public void processShouldGetAnnotationsAndReturnCompleteResponseWithoutTruncateD
                 .thenReturn(okResponse);
 
-        processor.process(annotationRequest, mockResponder, mockImapSession);
+        processor.process(annotationRequestBuilder.build(), mockResponder, mockImapSession);
 
         verify(mockMailboxManager, times(1)).getAnnotationsByKeysWithOneDepth(eq(inbox), eq(mailboxSession), eq(KEYS));
     }
 
     @Test
-    public void processShouldGetAnnotationsByKeysWithAllDepth() throws Exception {
-        annotationRequest.setKeys(KEYS);
-        annotationRequest.setDepthValue(Depth.INFINITY);
+    public void processShouldCallGetAnnotationsByKeysWithAllDepth() throws Exception {
+        annotationRequestBuilder.keys(KEYS).depth(Depth.INFINITY).build();
 
         when(mockStatusResponseFactory.taggedOk(any(String.class), any(ImapCommand.class), any(HumanReadableText.class)))
                 .thenReturn(okResponse);
 
-        processor.process(annotationRequest, mockResponder, mockImapSession);
+        processor.process(annotationRequestBuilder.build(), mockResponder, mockImapSession);
 
         verify(mockMailboxManager, times(1)).getAnnotationsByKeysWithAllDepth(eq(inbox), eq(mailboxSession), eq(KEYS));
     }
 
+    @Test
+    public void processShouldGetAnnotationsAndReturnCompleteResponseWithTruncateDataByMaxsize() throws Exception {
+        annotationRequestBuilder.maxsize(15);
+        when(mockMailboxManager.getAllAnnotations(inbox, mailboxSession)).thenReturn(ImmutableList.of(SHARED_ANNOTATION, PRIVATE_ANNOTATION, PRIVATE_CHILD_ANNOTATION, PRIVATE_GRANDCHILD_ANNOTATION));
+        when(mockStatusResponseFactory.taggedOk(any(String.class), any(ImapCommand.class), any(HumanReadableText.class)))
+                .thenReturn(okResponse);
+
+        ArgumentCaptor<AnnotationResponse> responderArgumentCaptor = ArgumentCaptor.forClass(AnnotationResponse.class);
+        processor.process(annotationRequestBuilder.build(), mockResponder, mockImapSession);
+        verify(mockResponder, times(2)).respond(responderArgumentCaptor.capture());
+
+        AnnotationResponse resultAnnotation = responderArgumentCaptor.getAllValues().get(0);
+        assertThat(resultAnnotation.getMailboxAnnotations()).contains(PRIVATE_ANNOTATION, SHARED_ANNOTATION);
+    }
+
+    @Test
+    public void processShouldGetAnnotationsAndReturnCompleteResponseWithoutTruncateDataByMaxsizeWhenNoMoreOverSizeItem() throws Exception {
+        annotationRequestBuilder.maxsize(100);
+        when(mockMailboxManager.getAllAnnotations(inbox, mailboxSession)).thenReturn(ImmutableList.of(SHARED_ANNOTATION, PRIVATE_ANNOTATION, PRIVATE_CHILD_ANNOTATION, PRIVATE_GRANDCHILD_ANNOTATION));
+        when(mockStatusResponseFactory.taggedOk(any(String.class), any(ImapCommand.class), any(HumanReadableText.class)))
+                .thenReturn(okResponse);
+
+        processor.process(annotationRequestBuilder.build(), mockResponder, mockImapSession);
+        verify(mockResponder, times(2)).respond(captorAnnotationResponse.capture());
```

Comment:
Do you have other interactions with your mockResponder ?

I guess you have a missing verify on ok() response code.

Finish by a verifyNoMoreInteraction(mockResponder);